### ISSUE TITLE
A way to remove the list of interactable blocks in the config

### ISSUE THREAD

**Issue by blubbarbs** [kernitus/BukkitOldCombatMechanics#623]
Hello, something that I noticed about this plugin is that it uses a list of interactable blocks in the config in order to check whether to shield when right clicking a block or not. I made a plugin that replicated old PVP myself some time back, and I found a way to solve it without needing such a list, which becomes a hassle to maintain the more blocks are added to Minecraft. 

The PlayerInteractEvent has the potential to fire 2 times, one for each hand. If the player interacted with a block that is interactable, however, the event will not fire at all for the off-hand. Thus, a simple guard clause would be needed in the PlayerInteractEvent listener:

```java
if (event.hasBlock() && event.getHand() == EquipmentSlot.HAND) return;
```

This catches the right-click on an interactable block. If the block was non-interactable, the event would fire again for the off-hand, at which point you know that the block was non-interactable and the logic for bringing up the shield can proceed as normal. That's all, I hope that this can help.

**Comment by kernitus** [comment]
Interesting, are these methods available in the 1.9 spigot API too? Also what do you mean about having two events, that there's two of the same type of event fired if the block is interactable?

**Comment by detobel36** [comment]
> Interesting, are these methods available in the 1.9 spigot API too? Also what do you mean about having two events, that there's two of the same type of event fired if the block is interactable?

In 1.10, yes

**Comment by kernitus** [comment]
Hmm yes it seems both methods [are available in 1.9](https://spigotdocs.okaeri.cloud/1.9.4/org/bukkit/event/player/PlayerInteractEvent.html). I will look into this after I am done rewriting the damage calculation components.

**Comment by kernitus** [comment]
So I just tested this with 1.19 and it doesn't seem to be the case at all. If I right click on a block the PlayerInteractEvent will fire, `event.hasBlock()` will be true and the hand will be HAND, but the event is not fired again for the offhand (which is currently empty). @blubbarbs @detobel36 did I misunderstand something in the way you are handling this?
